js/src/util/index.js

```diff
@@ -34,6 +34,19 @@ const getUID = prefix => {
   return prefix
 }
 
+const getTargetIdFromElement = element => {
+  const selector = element.getAttribute('data-bs-target') || element.getAttribute('href')
+
+  if (selector) {
+    // a valid HTML5 id is any string without whitespace
+    const match = selector.match(/#(\S+)/)
+    const id = match ? match[1] : null
+    return document.getElementById(id) ? id : null
```

Comment:
Can't this be tweaked a bit further? Like, there's no need to even call `getElementById` if `id` is `null`.

Reply:
True, but `getElementById()` already performs a check against `null` anyway.

**As-is**
`getElementById()` always performs a single check on whether `id` is null.

**Your proposed tweak**
If `id` is null, then we perform a single check on whether it's null. This is probably slightly faster than the function call to `getElementById()`.
If `id` is not null, then we still perform a check ourselves, but then `getElementById()` also performs a check. In other words, there are two branching statements.

In other words, your proposal may speed up the case when `id` is null, but it's slower when `id` is not null. I'm happy with either option. Feel free to change it as you see fit.